### PR TITLE
openstack: implement a periodic job to test availability zones

### PR DESCRIPTION
**Pull request by EmilienM**
Note: Inspiration from the AWS jobs which already deploy clusters in specific
AZs.

OpenStack cloud can also have multiple availability zones and we expose
it to our users via options in install-config.yaml.

We have some known limitations but we'll be working on them in the
future:
* Only work in BYON mode, because the router has to be
  created under a specific AZ. The installer doesn't do it for us yet.
* The cluster has to live within a single AZ for now, we have not tested
  stretched OCP clusters across multiple OpenStack AZ.

This patch creates a new periodic job, named `e2e-openstack-az` running on the
new `mecha` cloud which has an `az0` availability zone, used by this
job.
We introduced two new variables:
* ZONES: defined per OpenStack cloud
* ZONES_COUNT: on how many AZs do we want to deploy our OCP cluster (1
  for now).

The `openstack-provision-machinesubnet-commands.sh` script now supports
availability zone(s) and will configure the router accordingly.

The `openstack-conf-generateconfig-commands.sh` script now supports
availability zone(s) and will configure the controlPlane & compute
machine pools accordingly.

Jira: [OSASINFRA-2373](https://issues.redhat.com/browse/OSASINFRA-2373)

### CODE REVIEW

**Review by pierreprinetti — COMMENTED** [review]
Do we want to

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 1]
```diff
@@ -19217,17 +19217,6 @@ periodics:
...
-    slack:
-      channel: '#assisted-deployment-ci'
-      job_states_to_report:
-      - failure
-      - error
-      report_template: '{{if eq .Status.State "success"}} :green_jenkins_circle: Job
-        *{{.Spec.Job}}* ended with *{{.Status.State}}*. <{{.Status.URL}}|View logs>
-        :green_jenkins_circle: {{else}} :red_jenkins_circle: Job *{{.Spec.Job}}* ended
-        with *{{.Status.State}}*. <{{.Status.URL}}|View logs> :red_jenkins_circle:
-        {{end}}'
```
Comment (pierreprinetti) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:19230]: Will this unrelated change go away with a rebase?

Comment (EmilienM) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:19230]: yes, hopefully. I don't know why it was removed when doing `make jobs`...

Comment (EmilienM) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:19230]: yeah it was cleaned when i re-ran the command.

**Comment by EmilienM** [comment]
@hasbro17 FYI previous failures were expected:
* CSI job has failing tests, we're tracking it on https://bugzilla.redhat.com/show_bug.cgi?id=1917710
* openstack-ccm is AFIK not working
* upgrade jobs are broken in CI (although upgrades work), we're also on it.

**Comment by hasbro17** [comment]
@EmilienM Thanks for clarifying. Other than that, SGTM.

**Review by hasbro17 — APPROVED** [review]
/approve
